backend-diagram-converter/core/src/main/java/org/camunda/community/migration/converter/visitor/impl/ConditionExpressionVisitor.java

```diff
@@ -51,12 +51,19 @@ public String localName() {
 
   @Override
   protected void visitBpmnElement(DomElementVisitorContext context) {
+    String expression = context.getElement().getTextContent();
     ExpressionTransformationResult transformationResult =
-        ExpressionTransformer.transform(context.getElement().getTextContent());
+        ExpressionTransformer.transform(expression);
     context.addConversion(
         SequenceFlowConvertible.class,
         conversion -> conversion.setConditionExpression(transformationResult.getNewExpression()));
-    context.addMessage(MessageFactory.conditionExpression(transformationResult));
+    if (transformationResult.hasExecution()) {
+      context.addMessage(MessageFactory.conditionExpressionExecution(transformationResult));
+    } else if (transformationResult.hasMethodInvocation()) {
```

Comment:
This should be an if-logic without else as the cases do not exclude each other. This would lead to potentially multiple messages added per expression, but the result is deterministic.

Reply:
Why should we bother the user with two or even three messages about the same topic?
```
TASK: Transformed expression '$execution.getVariable("a") != null' -> '=execution.getVariable("a") != null'. 'execution' is not available in FEEL
TASK: Transformed expression '$execution.getVariable("a") != null' -> '=execution.getVariable("a") != null'. Method invocation is not available in FEEL
REVIEW: Please review transformed expression '$execution.getVariable("a") != null' -> '=execution.getVariable("a") != null'.
```

There is no additional benefit of this duplication.

The above-mentioned pattern is often used, and it could be transformed automatically, similar to #153.

The mapped result should be `=a != null`.

Reply:
Ok I see, let's leave it like it is